regression/dataset_mixing.yaml

```diff
@@ -0,0 +1,91 @@
+name: streaming-regression-test-dataset-mixing
+compute:
+  gpus: 8 # Number of GPUs to use
+  # cluster: TODO # Name of the cluster to use for this run
+command:
+  "pip uninstall -y mosaicml-streaming &&
+  cd streaming &&
+  pip install -e '.[dev]'
+
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_0/ --num-samples 5000 --size_limit 4096 --seed 0 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_1/ --num-samples 5000 --size_limit 4096 --seed 1 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_2/ --num-samples 5000 --size_limit 4096 --seed 2 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_3/ --num-samples 5000 --size_limit 4096 --seed 3 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_4/ --num-samples 5000 --size_limit 4096 --seed 4 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_5/ --num-samples 5000 --size_limit 4096 --seed 5 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_6/ --num-samples 5000 --size_limit 4096 --seed 6 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_7/ --num-samples 5000 --size_limit 4096 --seed 7 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_8/ --num-samples 5000 --size_limit 4096 --seed 8 &&
+  python regression/synthetic_dataset.py --create --name numberandsaydataset --out /tmp/streaming_dataset/stream_9/ --num-samples 5000 --size_limit 4096 --seed 9 &&
+
+  # Check sample order with stream choose
+
+  composer -n 1 regression/iterate_data.py --num_canonical_nodes 128 --batch_size 1
```

Comment:
Can we use some higher batch size such as `16` or `32` instead of `1` since `batch_size` of `1` does not touch certain partition logic ? Same comments for every composer call in this file.